tests/php/Domain/Services/CreateAccount.php

```diff
@@ -19,9 +20,9 @@
  * @since $VID:$
  */
 class CreateAccount extends WP_UnitTestCase {
-
 	private function get_test_instance() {
-		return new TestedCreateAccount( new Package( 'test', './' ) );
+		Package::init();
```

Comment:
I needed to manually init `Package` in order to get this test to pass, otherwise I was getting:

`Exception: Cannot construct an instance of Automattic\WooCommerce\Blocks\Domain\Services\FeatureGating because it has not been registered.`

I saw there was a discussion between @nerrad and @senadir earlier in Slack about a similar error, but I couldn't figure out how to solve it in this specific case. My idea was to inject `FeatureGating` into `Package` in order to avoid [getting it directly from the container](https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/3343/files#diff-deccd9a660832b41efc10cb868f4549a32152b0e3753555c179b84a4733ee7cbR81), but that doesn't seem to be possible because `Package` is constructed before `FeatureGating` is registered. Any clues?

Reply:
Once you implement my request above to inject `FeaturePlugin` you'd be able to implement this via:

```
return new TestedCreateAccount( new Package( 'test', './', new FeaturePlugin( 3 ) );
```

That's one of the reasons for injecting dependencies because then mocking is much easier. As noted in the earlier comment, anywhere `Package` is instantiated directly will have to be updated to implement the new signature. Another benefit of this is that we are explicitly aware when testing what gating is in effect for the test.